common/define-grammar.js

```diff
@@ -683,9 +711,29 @@ module.exports = function defineGrammar(dialect) {
         prec(PREC.ARRAY_TYPE+1, seq($._primary_type, '[', ']'))
       )),
 
+      _tuple_type_body: $ => seq(
+        "[",
+        optional(
+          seq(
+            $._tuple_type_member,
+            optional(
+              seq(
+                optional(
+                  seq(
```

Comment:
It looks like there's some redundant nested `optional` and `seq` calls in here. Can it just be

```js
      _tuple_type_body: $ => seq(
        "[",
        optional(seq(
          commaSep1($._tuple_type_member),
          optional(',')
        )),
        "]"
      ),
```

Reply:
I managed to reduce this further. Please check again.